src/Raven.Studio/typescript/viewmodels/database/documents/DeleteDocumentsModal.tsx

```diff
@@ -0,0 +1,193 @@
+        async () => {
+            const collectionNameForApi = currentCollection().isAllDocuments ? "@all_docs" : currentCollection().name;
+            const result = await databasesService.deleteCollection(collectionNameForApi, dbName, excludedIds);
+            return result.OperationId;
+        },
+        {
+            onError: (error) => messagePublisher.reportError("Failed to delete collection", error.message),
+        },
+    );
+    
+    const monitorOperationAsync = useAsyncCallback(
+        async (operationId: number) => {
+            notificationCenter.instance.openDetailsForOperationById(dbName, operationId);
+            close();
+            await notificationCenter.instance.databaseOperationsWatch.monitorOperation(operationId)
+                .done(() => {
+                    if (excludedIds.length === 0) {
+                        messagePublisher.reportSuccess(`Deleted collection ${currentCollection().name}`);
+                    } else {
+                        messagePublisher.reportSuccess(`Deleted ${pluralizeHelpers.pluralize(documentCount, "document", "documents")} from ${currentCollection().name}`);
+                    }
+                    
+                    if (excludedIds.length === 0) {
+                        // if entire collection was deleted then go to 'all documents'
+                        const allDocsCollection = collectionsTracker.default.getAllDocumentsCollection();
+                        if (currentCollection() !== allDocsCollection) {
+                            currentCollection(allDocsCollection);
+                        }
+                    }
+                })
+                .always(() => onDeleteCompleted());
+        },
+        {
+            onError: (error) => messagePublisher.reportError("Failed to monitor delete operation", error.message),
+        },
+    );
+    
+    const execute = async () => {
+        const operationId = await deleteCollectionAsync.execute();
+        if (operationId) {
+            await monitorOperationAsync.execute(operationId);
+        }
+    };
+    
+    return {
+        execute,
+        loading: deleteCollectionAsync.loading || monitorOperationAsync.loading,
+    };
+}
+
+interface CollectionsInfoProps {
+    virtualGridSelection: virtualGridSelection<document>;
+    collectionsList: UseAsyncReturn<collectionsStats>;
+    currentCollection: KnockoutObservable<collection>;
+}
+
+function CollectionsInfo({
+                             virtualGridSelection,
+                             collectionsList,
+                             currentCollection,
+                         }: CollectionsInfoProps) {
+    const collectionName = currentCollection().name === collection.allDocumentsCollectionName ? "all" : currentCollection().name;
+    return (
+        <LazyLoad active={collectionsList.loading}>
+            <div>
+                <p>
+                    All documents from <b>{collectionName}</b> collections will be deleted
```

Comment:
* Text is fine when referring to 'all' collections.
   (Although, imho, would be better to use "ALL" instead of "all").
* In the case where it is a 'single collection', the resulting text here would be:
"All documents from &lt;collectionName&gt; collections will be deleted" 
=> but it should be:
"All documents from collection &lt;collectionName&gt; will be deleted.
* See the comment in:
https://issues.hibernatingrhinos.com/issue/RavenDB-21975/Add-approve-checkbox-on-deleting-all-documents#focus=Comments-67-1064488.0-0